docs/workflows/create_workflow_block.md

```diff
@@ -0,0 +1,1898 @@
+            description="Selectors to step outputs",
+            examples=[{"a": $steps.model_1.predictions", "b": "$Inputs.data"}],
+        )
+    
+        @classmethod
+        def describe_outputs(cls) -> List[OutputDefinition]:
+            return [
+              OutputDefinition(name="my_output", kind=[]),
+            ]
+    
+        @classmethod
+        def get_execution_engine_compatibility(cls) -> Optional[str]:
+            return ">=1.0.0,<2.0.0"
+    
+    
+    class BlockWithNamedSelectorsV1(WorkflowBlock):
+    
+        @classmethod
+        def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+            return BlockManifest
+    
+        def run(
+            self,
+            data: Dict[str, Any],
+        ) -> BlockResult:
+            ...
+            return {"my_output": ...}
+    ```
+
+    * lines `23-26` depict how to define manifest field capable of accepting 
+    list of selectors
+
+    * line `47` shows what to expect as input to block's `run(...)` method - 
+    dict of objects which are reffered with selectors. If the block accepted 
+    batches, the input type of `data` field would be `Dict[str, Union[Batch[Any], Any]]`.
+    In non-batch cases, non-batch-oriented data referenced by selector is automatically 
+    broadcasted, whereas for blocks accepting batches - `Batch` container wraps only 
+    batch-oriented inputs, with other inputs being passed as singular values.
+
+Such block is compatible with the following step declaration:
+
+```{ .json linenums="1" hl_lines="4-7"}
+{
+  "type": "my_plugin/named_selectors_example@v1",
+  "name": "my_step",
+  "data": {
+    "a": "$steps.model_1.predictions",
+    "b": "$inputs.my_parameter"  
+  }
+}
+```
+
+Practical implications will be the following:
+
+* under `data["a"]` inside `run(...)` you will be able to find model's predictions - 
+like `sv.Detections` if `model_1` is object-detection model
+
+* under `data["b"]` inside `run(...)`, you will find value of input parameter named `my_parameter`
+
+### Inputs and output dimensionality vs `run(...)` method
+
+The dimensionality of block inputs plays a crucial role in shaping the `run(...)` method’s signature, and that's 
+why the system enforces strict bounds on the differences in dimensionality levels between inputs 
+(with the maximum allowed difference being `1`). This restriction is critical for ensuring consistency and 
+predictability when writing blocks.
+
+If dimensionality differences weren't controlled, it would be difficult to predict the structure of 
+the `run(...)` method, making development harder and less reliable. That’s why validation of this property 
+is strictly enforced during the Workflow compilation process.
+
+Similarly, the output dimensionality also affects the method signature and the format of the expected output. 
+The ecosystem supports the following scenarios:
+
+* all inputs have **the same dimensionality** and output **do not change** dimensionality - baseline case
```

Comment:
`and output **do not change** dimensionality`

either:
`and outputs **do not change** dimensionality`
or
`and output **does not change** dimensionality`

Reply:
👍